aws/data_source_aws_elasticache_replication_group_test.go

```diff
@@ -0,0 +1,94 @@
+			{
+				Config: testAccDataSourceAwsElasticacheReplicationGroupConfig_basic(rName),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr("data.aws_elasticache_replication_group.bar", "replication_group_id", fmt.Sprintf("tf-%s", rName)),
+					resource.TestCheckResourceAttr("data.aws_elasticache_replication_group.bar", "replication_group_description", "test description"),
+					resource.TestCheckResourceAttr("data.aws_elasticache_replication_group.bar", "auth_token_enabled", "false"),
+					resource.TestCheckResourceAttr("data.aws_elasticache_replication_group.bar", "automatic_failover_enabled", "true"),
+					resource.TestCheckResourceAttr("data.aws_elasticache_replication_group.bar", "port", "6379"),
+					resource.TestCheckResourceAttrSet("data.aws_elasticache_replication_group.bar", "primary_endpoint_address"),
+					resource.TestCheckResourceAttr("data.aws_elasticache_replication_group.bar", "number_cache_clusters", "2"),
+					resource.TestCheckResourceAttr("data.aws_elasticache_replication_group.bar", "node_type", "cache.m1.small"),
+					resource.TestCheckResourceAttr("data.aws_elasticache_replication_group.bar", "snapshot_window", "01:00-02:00"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccDataSourceAwsElasticacheReplicationGroup_ClusterMode(t *testing.T) {
+	rName := acctest.RandString(10)
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceAwsElasticacheReplicationGroupConfig_ClusterMode(rName),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr("data.aws_elasticache_replication_group.cluster", "replication_group_id", fmt.Sprintf("tf-%s", rName)),
+					resource.TestCheckResourceAttr("data.aws_elasticache_replication_group.cluster", "replication_group_description", "test description"),
+					resource.TestCheckResourceAttr("data.aws_elasticache_replication_group.cluster", "auth_token_enabled", "false"),
+					resource.TestCheckResourceAttr("data.aws_elasticache_replication_group.cluster", "automatic_failover_enabled", "true"),
+					resource.TestCheckResourceAttr("data.aws_elasticache_replication_group.cluster", "port", "6379"),
+					resource.TestCheckResourceAttrSet("data.aws_elasticache_replication_group.cluster", "configuration_endpoint_address"),
+					resource.TestCheckResourceAttr("data.aws_elasticache_replication_group.cluster", "node_type", "cache.m1.small"),
+				),
+			},
+		},
+	})
+}
+
+func testAccDataSourceAwsElasticacheReplicationGroupConfig_basic(rName string) string {
+	return fmt.Sprintf(`
+resource "aws_elasticache_replication_group" "bar" {
+	replication_group_id = "tf-%s"
+	replication_group_description = "test description"
+	node_type = "cache.m1.small"
+	number_cache_clusters = 2
+	port = 6379
+	parameter_group_name = "default.redis3.2"
+	availability_zones = ["us-east-1a", "us-east-1b"]
```

Comment:
This is currently failing in our AWS testing account:

```
=== RUN   TestAccDataSourceAwsElasticacheReplicationGroup_basic
--- FAIL: TestAccDataSourceAwsElasticacheReplicationGroup_basic (21.10s)
	testing.go:492: Step 0 error: Error applying: 1 error(s) occurred:

		* aws_elasticache_replication_group.bar: 1 error(s) occurred:

		* aws_elasticache_replication_group.bar: Error creating Elasticache Replication Group: InvalidParameterValue: us-east-1b is not a valid availability zone.
			status code: 400, request id: d01f865d-c572-11e7-8067-e323f32a1242
```
and it's usually not a good practice to hard-code AZs.

Can you please make it reusable by using the AZ data source here? https://www.terraform.io/docs/providers/aws/d/availability_zones.html